extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/job/Job.java

```diff
@@ -111,8 +131,15 @@ public static void scheduleRange(String thingUID, AstroThingHandler astroHandler
         if (thingNull || astroHandlerNull || rangeNull || channelIdNull) {
             return;
         }
-        scheduleEvent(thingUID, astroHandler, range.getStart(), EVENT_START, channelId);
-        scheduleEvent(thingUID, astroHandler, range.getEnd(), EVENT_END, channelId);
+        
+        Calendar start = range.getStart();
+        Calendar end = range.getEnd();
+        if (truncatedEquals(start, end, SECOND)) {
+            scheduleEvent(thingUID, astroHandler, start, asList(EVENT_START, EVENT_END), channelId);
+            return;
```

Comment:
please use an `if {...} else {...}` block here - the early return is of course correct, but always dangerous for the next one who needs to do some archeology here (or even change it).